universal-login-sdk/lib/api/FutureWalletFactory.ts

```diff
@@ -56,16 +91,19 @@ export class FutureWalletFactory {
     const deploy = async (ensName: string, gasPrice: string, gasToken: string) => {
       const initData = await this.setupInitData(publicKey, ensName, gasPrice, gasToken);
       const signature = await calculateInitializeSignature(initData, privateKey);
-      await this.relayerApi.deploy(publicKey, ensName, gasPrice, gasToken, signature, this.sdk.sdkConfig.applicationInfo);
-
-      const deploymentObserver = new DeploymentObserver(this.blockchainService, this.config.contractWhiteList);
-      const address = await new Promise<string>((resolve) => deploymentObserver.startAndSubscribe(contractAddress, resolve));
-      return new DeployedWallet(
-        address,
-        ensName,
-        privateKey,
-        this.sdk,
-      );
+      const deploymentHash = await this.relayerApi.deploy(publicKey, ensName, gasPrice, gasToken, signature, this.sdk.sdkConfig.applicationInfo);
+
+      const deployment: Deployment = {
+        waitForTransactionHash: this.createWaitForTransactionHash(deploymentHash),
+        waitToBeSuccess: this.createWaitToBeSuccess(deploymentHash),
+        deployedWallet: new DeployedWallet(
+          contractAddress,
+          ensName,
+          privateKey,
+          this.sdk,
+        )
+      };
+      return deployment;
```

Comment:
This is breaking change - it changes the behaviour of existing SDK's `deploy` function.